### ISSUE TITLE
INCR support

### ISSUE THREAD

**Issue by vejkse** [cdown/clipmenu#235]
With clipmenu 7 (that I start with `CM_SELECTIONS=clipboard` and `CM_OWN_CLIPBOARD=1`, but the latter option has no influence on what follows), if I run:
```bash
printf '%.0sa' {1..4000} | xsel --clipboard
```
I get, as expected, a 4000 *a*s long string in the clipboard.  On the other hand, if I run:
```bash
printf '%.0sa' {1..4001} | xsel --clipboard
```
I get a few random non-printable bytes in the clipboard.

A work around is using xclip:
```bash
printf '%.0sa' {1..4001} | xclip -selection clipboard
```
works fine.

I am not entirely sure about the responsibility.  There was a bug in xsel that was fixed a few years ago but released only last year (version 1.2.1); it may still not be completely fixed: https://github.com/kfish/xsel/issues/33 so maybe there is some part of responsibility with xsel.

But with clipmenu 6 and the same version of xsel, there was no problem.  And the reason for the 4000 threshold in the xsel bugs is that xsel uses the INCR protocol for more than 4000 bytes (xclip uses a different method).  Maybe clipmenu 7 has trouble with that protocol?

**Comment by cdown** [comment]
Yeah, we don't have INCR support yet (it's written, but I am not happy with the state of the code yet), that's one of the reasons clipmenu 7 isn't released yet. Thanks for the report!

**Comment by cdown** [comment]
I was wondering why I'm not experiencing this: I guess xclip's limits for using INCR are higher.

**Comment by vejkse** [comment]
Apparently, for xclip the limit is at 1 MB, more exactly between
```bash
printf '%.0sa' {1..1048575} | xclip -selection clipboard
```
and
```bash
printf '%.0sa' {1..1048576} | xclip -selection clipboard
```

**Comment by cdown** [comment]
INCR support is in #236. Testing appreciated, I'll merge in a few days if I don't hear back :-)

**Comment by vejkse** [comment]
I just tried to test it but couldn’t get there.  Maybe I did something wrong.  (I replaced `develop` with `cdown/2024-11-08/incr` in the PKGBUILD of clipmenu-git: https://aur.archlinux.org/packages/clipmenu-git and built that modified package.)

When trying to run `env CM_SELECTIONS=clipboard CM_OWN_CLIPBOARD=1 clipmenud` (after having killed the previously running `clipmenud` and removed the previous `clipmenu.7.1000` directory), two things happen:

1. The directory `clipmenu.7.1000` is recreated with 2 or 3 clips: the current content of the clipboard and, bizarrely, the value of some properties of the X root window: usually `_XMONAD_LOG` and once also `RESOURCE_MANAGER`.
2. `clipmenud` quits with the error `FATAL: Unreachable`.

**Comment by cdown** [comment]
Well, that's not good! I don't repro, but I believe I worked out the problem -- could you please refetch the branch, rebuild, and run with CM_DEBUG=1?

**Comment by vejkse** [comment]
Now it works without any problem, including the INCR with either xsel or xclip.  Do you need the output with `CM_DEBUG=1`, since there is no error?

**Comment by cdown** [comment]
That's great! Then it's fine. Thanks for testing.

**Comment by cdown** [comment]
Merged in #236.
